Add justknobs for static cuda launcher

jamesjwu opened the pull request:
Summary:
This diff adds a justknobs check for static cuda launcher. In particular, it supports a fractional rollout where each mast job/version can be consistently enrolled in the config on or off. 

It also adds a set_feature_use so we can track whether static cuda launcher is enabled on a given dynamo compile.

Test Plan: Existing unit tests. The justknobs in question are set to be disabled right now, so this diff does not launch the feature yet.

Differential Revision: D74599203




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

oulgen commented on torch/_inductor/config.py:
```diff
@@ -40,11 +40,35 @@ def bundle_triton_into_fx_graph_cache_default() -> Optional[bool]:
+        if STATIC_CUDA_LAUNCHER_VERSION < version:
+            return False
+
+        # Next, we do a fractional rollout on mast jobs
```
@c00w are we ok with a rollout like this?

c00w replied:
Yeah it's fine, I really want to fix the infra so we don't have to do this dance, but haven't done it yet.

jamesjwu committed:
Add justknobs for static cuda launcher (#153400)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/153400

This diff adds a justknobs check for static cuda launcher. In particular, it supports a fractional rollout where each mast job/version can be consistently enrolled in the config on or off.

It also adds a set_feature_use so we can track whether static cuda launcher is enabled on a given dynamo compile.

Test Plan: Existing unit tests. The justknobs in question are set to be disabled right now, so this diff does not launch the feature yet.

Reviewed By: masnesral

Differential Revision: D74599203
